RFC/0001-graph-display.md

```diff
@@ -0,0 +1,241 @@
+# WIP: Graph Display Architecture for QML
+
+**Authors:**
+
+- Adam Washington
+
+## 1 Executive Summary
+
+This document specifies an architecture for a graph display library
+for the Dissolve QML GUI and requests discussion and suggestions for
+improvement.
+
+## 2 Motivation
+
+Our current system for creating layers and generators has multiple issues with the user experience:
+
+- Modules are presented in a linear list for what might be a highly
+  parallel process.
+- It is only possible to examine one module at a time.
+- Values shared between nodes are contained in a keyword system that
+  can be both opaque (hard to determine keyword values) and brittle
+  (long but similar keyword names that must be typed exactly).
+- It is difficult for users to create complicated layers without
+  outside help.
+- There is no method of gaining a high level overview of the
+  activities of a layer.  Each module much be read in sequence and a
+  mental modal of the entire, possibly non-linear, process must be
+  constructed from this.
+  
+A graphical display of a connected nodes provides an alternative that
+ameliorates most of these issues.
+
+- All modules can be seen at once with the relations between modules
+  explicitly declared by the connection.
+- All modules and their values are present on the workspace at the same time.
+- Values are shared by drawing connections between modules instead of
+  variables from a global namespace.
+- The graphical interface provides an intuitive framework for treating
+  the modules as individual building blocks.
+- One glance at the workflow gives an overview of all of the modules
+  and their relations
+
```

Comment:
- Links between data / objects is well-defined and strong, removing data management issues almost completely